sympy/solvers/solveset.py

```diff
@@ -314,6 +311,81 @@ def _invert_complex(f, g_ys, symbol):
     return (f, g_ys)
 
 
+def _invert_abs(f, g_ys, symbol):
+    """Helper function for inverting absolute value functions.
+
+    Returns the complete result of inverting an absolute value function.
+    Alongwith finding all the possible solutions of the given function,
+    the routine determines a set of conditions which must be satified
+    for their existence.
+
+    The returned values are determined by the condition
+    If it is certain that all these conditions are met, a `FiniteSet` of all
+    possible solutions is returned.
+    If it can be concluded that atleast one condition cannot be satified, an
+    `EmptySet` is returned.
+    In all other cases, a `ConditionSet` of the solutions, with all the required
+    conditions specified, is returned.
+
+    """
+
+    n = Dummy('n', real=True)
+
+    conditions = S.EmptySet
+    condition_interval = Interval(0, oo)
+
+    if f.is_positive:
+        f_n = f
+    else:
+        f_n = f.args[0]
+    g_x, values = _invert_real(f_n, Union(imageset(Lambda(n, n), g_ys),
+                               imageset(Lambda(n, -n), g_ys)), symbol)
+
+    if isinstance(values, ConditionSet):
```

Comment:
@smichr I might be wrong but I don't understand this. Why we need to check for `ConditionSet`, `_invert_real` never returns a `ConditionSet`, hence this will never be True.

Reply:
I suspect it does but can check. The reason I think it does is because I had to watch out for the Contains using evaluate True or False (before making it so Contains no longer returns a Relational).

But I didn't write the code -- I just did the minimal amount to get #10751 up and running again.

Reply:
I pruned out some code. Let's see if tests pass.